internal/manifests/cloudcredentials/credentialsrequests.go

```diff
@@ -0,0 +1,133 @@
+package cloudcredentials
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	cloudcredentialv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/grafana/tempo-operator/api/tempo/v1alpha1"
+	"github.com/grafana/tempo-operator/internal/manifests"
+	"github.com/grafana/tempo-operator/internal/manifests/manifestutils"
+)
+
+type k8client interface {
+	Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error
+	Delete(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error
+	CreateOrUpdate(ctx context.Context, obj client.Object, f controllerutil.MutateFn) (controllerutil.OperationResult, error)
+}
+
+// CredentialRequestOptions options for creating the credential request CCO CR.
+type CredentialRequestOptions struct {
+	TokenCCOAuth   manifestutils.TokenCCOAuthConfig
+	ServiceAccount string
+	Controlled     metav1.Object
+	CredentialMode v1alpha1.CredentialMode
+}
+
+// CreateUpdateDeleteCredentialsRequest creates a new CredentialsRequest resource for a TempoStack
+// to request a cloud credentials Secret resource from the OpenShift cloud-credentials-operator.
+func CreateUpdateDeleteCredentialsRequest(ctx context.Context, scheme *runtime.Scheme,
+	params CredentialRequestOptions, k k8client) error {
+
+	if params.CredentialMode != v1alpha1.CredentialModeTokenCCO {
```

Comment:
Is this condition only required for pruning, i.e. switching from CCO back to non-CCO?
When we use `reconcileManagedObjects`, we'd get the pruning functionality for free.